### PR TITLE
cache clearing now with sudo

### PR DESCRIPTION
**Pull request by SanJSp**
THIS MAKES SUDO RIGHTS MANDATORY TO RUN BENCHMARKS!

After this discussion with Stefan and Martin, we figured out, that we aren't clearing our caches. This should fix it.

See: https://hyrise-db.slack.com/archives/C04G68N1VLK/p1673340099481499

### CODE REVIEW

**Comment by tjjordan** [comment]
Here are some benchmarks results created with fixed cache clearing.
They probably aren't reproducible because they're from the VM.
Bildschirm­foto 2023-01-10 um 11 33 47

File: src/benchmark/micro_benchmark_utils.cpp [hunk 1]
```diff
@@ -23,8 +25,11 @@ void micro_benchmark_clear_cache() {
 void micro_benchmark_clear_disk_cache() {
   // TODO(phoenix): better documentation of which caches we are clearing
   sync();
-  std::ofstream ofs("/proc/sys/vm/drop_caches");
-  ofs << "3" << std::endl;
+#ifdef __APPLE__
+  system("purge");
```
Comment (phoeinx) [inline src/benchmark/micro_benchmark_utils.cpp:29]: looks good! short question - do you have any official documentation from MacOS or XCode etc. that defines what exactly `purge` does? I could only find secondary sources so far.

**Comment by tjjordan** [comment]
Same VM, without working cache clearing:
Bildschirm­foto 2023-01-10 um 12 20 22

**Comment by phoeinx** [comment]
Wow, caching strikes again. Thanks for the measurements!
You don't by any chance have csvs for the runs so that we can quickly create some graphs?
It appears like this might really change our previous results and trends in the results:

E.g. the impact of threading on PREAD RANDOM:
with proper cache clearing -> threads show positive impact on performance
image

w.o. proper caching clearing -> threads show negative impact on performance
image

File: src/benchmark/micro_benchmark_utils.cpp (hunk 1 above)
Comment (SanJSp) [reply src/benchmark/micro_benchmark_utils.cpp:29]: Maybe man purge is ok?

https://stackoverflow.com/a/13659800

PURGE(8)                                                                                 System Manager's Manual                                                                                PURGE(8)

NAME
     purge – force disk cache to be purged (flushed and emptied)

SYNOPSIS
     purge

DESCRIPTION
     Purge can be used to approximate initial boot conditions with a cold disk buffer cache for performance analysis.  It does not affect anonymous memory that has been allocated through malloc,
     vm_allocate, etc.

SEE ALSO
     malloc(3), sync(8)

Comment (tjjordan) [reply src/benchmark/micro_benchmark_utils.cpp:29]: https://stackoverflow.com/questions/28845524/echo-3-proc-sys-vm-drop-caches-on-mac-osx
This was my secondary source ¯\_(ツ)_/¯

**Pull request merged**